pkg/adapter/config/encryptionAdapter.go

```diff
@@ -2,29 +2,129 @@ package config
 )
 
 type PublicKeyConfigEncryptionAdapter struct {
+	secrets   secret
+	registry  etcdRegistry
+	namespace string
 }
 
-func NewPublicKeyConfigEncryptionAdapter() *PublicKeyConfigEncryptionAdapter {
-	return &PublicKeyConfigEncryptionAdapter{}
+func NewPublicKeyConfigEncryptionAdapter(secretClient secret, registry etcdRegistry, namespace string) *PublicKeyConfigEncryptionAdapter {
+	return &PublicKeyConfigEncryptionAdapter{secrets: secretClient, registry: registry, namespace: namespace}
 }
 
 func (p PublicKeyConfigEncryptionAdapter) Encrypt(
 	ctx context.Context,
 	name common.SimpleDoguName,
 	value common.SensitiveDoguConfigValue,
 ) (common.EncryptedDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+	pubkey, err := resource.GetPublicKey(p.registry, string(name))
+	if err != nil {
+		return "", &domainservice.NotFoundError{
+			WrappedError: err,
+			Message:      "could not get public key",
+		}
+	}
+	encryptedValue, err := pubkey.Encrypt(string(value))
+	if err != nil {
+		return "", domainservice.NewInternalError(err, "could not encrypt value")
+	}
+	return common.EncryptedDoguConfigValue(encryptedValue), nil
 }
 
 func (p PublicKeyConfigEncryptionAdapter) EncryptAll(
 	ctx context.Context,
 	entries map[common.SensitiveDoguConfigKey]common.SensitiveDoguConfigValue,
 ) (map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue, error) {
-	//TODO implement me
-	panic("implement me")
+	encryptedEntries := map[common.SensitiveDoguConfigKey]common.EncryptedDoguConfigValue{}
+	var encryptionErrors []error
+	pubkeys := map[string]*keys.PublicKey{}
+	for configKey, configValue := range entries {
+		doguname := string(configKey.DoguName)
+		pubkey, pubkeyKnown := pubkeys[doguname]
+		if !pubkeyKnown {
+			pubkeyFromRegistry, err := resource.GetPublicKey(p.registry, doguname)
+			if err != nil {
+				encryptionErrors = append(encryptionErrors, domainservice.NewNotFoundError(err, "could not get public key for dogu %v", doguname))
+				continue
+			} else {
+				pubkey = pubkeyFromRegistry
+				pubkeys[doguname] = pubkeyFromRegistry
+			}
+		}
+		encryptedValue, err := pubkey.Encrypt(string(configValue))
+		if err != nil {
+			encryptionErrors = append(encryptionErrors, domainservice.NewInternalError(err, "could not encrypt value for dogu %v", doguname))
+		}
+		encryptedEntries[configKey] = common.EncryptedDoguConfigValue(encryptedValue)
+	}
+	return encryptedEntries, errors.Join(encryptionErrors...)
+}
+
+func (p PublicKeyConfigEncryptionAdapter) Decrypt(
+	ctx context.Context,
+	name common.SimpleDoguName,
+	encryptedValue common.EncryptedDoguConfigValue) (common.SensitiveDoguConfigValue, error) {
+	privateKeySecret, err := p.secrets.Get(ctx, string(name)+"-private", metav1.GetOptions{})
+	if err != nil {
+		return "", domainservice.NewNotFoundError(err, "could not get private key")
+	}
+	privateKey := privateKeySecret.Data["private.pem"]
+	keyPair, err := getKeyPairFromPrivateKey(privateKey, p.registry)
```

Comment:
🚫 **[checkstyle]** <ineffassign> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>ineffectual assignment to err

Reply:
I don't really understand why this is ineffectual as it shouldn't create a new variable and it's used afterwards.
According to https://github.com/gordonklaus/ineffassign, it shouldn't give any false-positives.

Reply:
Ah, I see now. You added check, but since you didn't change this line it didn't show as outdated.